src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/ProjectCapabilitiesMissingVetoProjectLoad.ProjectType.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    internal partial class ProjectCapabilitiesMissingVetoProjectLoad
+    {
+        private class ProjectType
```

Comment:
Could be a `readonly struct`.

Reply:
Could be but would result in basically unneeded JIT code gen for basically 3 instances on the heap.